### PR TITLE
[Silabs] Adds refrigerator app matter shell support for change door state

### PR DESCRIPTION
**Pull request by shgutte**
**Problem:** Unable to test the test case *TC-REFALM-2.2* due to the lack of functionality to open and close the refrigerator door.  
**Solution:** Added support in the Matter shell to modify the door state.  
**Tested On:** SiWx917 device with test case *TEFALM-2.2*.

### CODE REVIEW

**Comment by lboue** [comment]
Building this PR for BRD2703A fails

```bash
./scripts/examples/gn_silabs_example.sh ./examples/refrigerator-app/silabs/ ./out/refrigerator-app-shell BRD2703A chip_build_libshell=true
```

**Build error**
```
In file included from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/CodeUtils.h:34,
                 from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/core/Optional.h:32,
                 from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/app/ConcreteAttributePath.h:23,
                 from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/zzz_generated/app-common/app-common/zap-generated/cluster-objects.h:26,
                 from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.h:20,
                 from ../../../examples/refrigerator-app/silabs/include/EventHandlerLibShell.h:19,
                 from ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:18:
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp: In function 'CHIP_ERROR RefrigeratorDoorEventHandler(int, char**)':
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:26: error: format '%d' expects argument of type 'int', but argument 4 has type 'chip::app::Cluster                s::RefrigeratorAlarm::AlarmBitmap' [-Werror=format=]
  109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~
      |                                                                       |
      |                                                                       chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap
../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:469:70: note: in definition of macro 'ChipInternalLogImpl'
  469 |             chip::Logging::Log(chip::Logging::kLogModule_##MOD, CAT, MSG, ##__VA_ARGS__);                                          \
      |                                                                      ^~~
../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:110:40: note: in expansion of macro 'ChipInternalLog'
  110 | #define ChipLogProgress(MOD, MSG, ...) ChipInternalLog(MOD, PROGRESS, MSG, ##__VA_ARGS__)
      |                                        ^~~~~~~~~~~~~~~
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:5: note: in expansion of macro 'ChipLogProgress'
  109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
      |     ^~~~~~~~~~~~~~~
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:61: note: format string is defined here
  109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
      |                                                            ~^
      |                                                             |
      |                                                             int
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp: In function 'void EventWorkerFunction(intptr_t)':
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:30: error: format '%d' expects argument of type 'int', but argument 4 has type 'chip::app::Cluster                s::RefrigeratorAlarm::AlarmBitmap' [-Werror=format=]
  158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~
      |                                                                       |
      |                                                                       chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap
../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:469:70: note: in definition of macro 'ChipInternalLogImpl'
  469 |             chip::Logging::Log(chip::Logging::kLogModule_##MOD, CAT, MSG, ##__VA_ARGS__);                                          \
      |                                                                      ^~~
../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:110:40: note: in expansion of macro 'ChipInternalLog'
  110 | #define ChipLogProgress(MOD, MSG, ...) ChipInternalLog(MOD, PROGRESS, MSG, ##__VA_ARGS__)
      |                                        ^~~~~~~~~~~~~~~
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:9: note: in expansion of macro 'ChipLogProgress'
  158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
      |         ^~~~~~~~~~~~~~~
../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:56: note: format string is defined here
  158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
      |                                                       ~^
      |                                                        |
      |                                                        int
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[841/895] c++ obj/third_party/connectedhomeip/examples/platform/silabs/efr32-common.OTAConfig.cpp.o
```

**Comment by lboue** [comment]
Thanks. Building this PR for BRD2703A is OK now
```
08:54:32.691 -> matterCli> refrigeratoralarm event
08:54:41.724 ->   help            Usage : refrigeratoralarm event <subcommand>
08:54:41.770 ->   door-state-change Sends door state change event to Refrigerator app
08:54:41.770 -> Done

08:54:41.770 -> matterCli> refrigeratoralarm event door-state-change open
08:54:52.761 -> Done
```

File: examples/refrigerator-app/silabs/include/AppTask.h [hunk 1]
```diff
@@ -48,6 +48,7 @@
 #define APP_ERROR_CREATE_TIMER_FAILED CHIP_APPLICATION_ERROR(0x04)
 #define APP_ERROR_START_TIMER_FAILED CHIP_APPLICATION_ERROR(0x05)
 #define APP_ERROR_STOP_TIMER_FAILED CHIP_APPLICATION_ERROR(0x06)
+#define APP_ERROR_TOO_MANY_SHELL_ARGUMENTS CHIP_APPLICATION_ERROR(0x08)
```
Comment (jmartinez-silabs) [inline examples/refrigerator-app/silabs/include/AppTask.h:51]:
```suggestion
```

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]

File: examples/refrigerator-app/silabs/include/AppTask.h (hunk 1 above)
Comment (shgutte) [reply examples/refrigerator-app/silabs/include/AppTask.h]: Deleted

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    sShellRefrigeratorSubCommands.RegisterCommands(sRefrigeratorSubCommands, ArraySize(sRefrigeratorSubCommands));
+
+    Engine::Root().RegisterCommands(&sRefrigeratorCommand, 1);
+
+    return CHIP_NO_ERROR;
+}
+
+void EventWorkerFunction(intptr_t context)
+{
+    VerifyOrReturn(context != 0, ChipLogError(NotSpecified, "EventWorkerFunction - Invalid work data"));
```
[the remark this answers is not in the document]
Comment (shgutte) [reply examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp]: Updated

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+{
+
+    if (argc == 0)
+    {
+        return AlarmHelpHandler(argc, argv);
+    }
+    if (argc >= 2)
+    {
+        ChipLogError(Zcl, "Too many arguments provided to function %s, line %d", __func__, __LINE__);
+        return APP_ERROR_TOO_MANY_SHELL_ARGUMENTS;
```
[the remark this answers is not in the document]
Comment (shgutte) [reply examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp]: replaced with CHIP_ERROR_INVALID_ARGUMENT

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    RefrigeratorAlarmEventData * data = Platform::New<RefrigeratorAlarmEventData>();
+    data->eventId         = Events::Notify::Id;
+    data->doorState      = static_cast<AlarmBitmap>(atoi(argv[0]));
```
[the remark this answers is not in the document]
Comment (shgutte) [reply examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp]: Added checks for integer

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    {
+        return AlarmHelpHandler(argc, argv);
+    }
+    if (argc >= 2)
+    {
+        ChipLogError(Zcl, "Too many arguments provided to function %s, line %d", __func__, __LINE__);
+        return APP_ERROR_TOO_MANY_SHELL_ARGUMENTS;
+    }
+
+    RefrigeratorAlarmEventData * data = Platform::New<RefrigeratorAlarmEventData>();
```
[the remark this answers is not in the document]
Comment (shgutte) [reply examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:115]: Handled the data leak in EventWorkerFunction

**Comment by shgutte** [comment]
> Building this PR for BRD2703A fails
> 
> ```shell
> ./scripts/examples/gn_silabs_example.sh ./examples/refrigerator-app/silabs/ ./out/refrigerator-app-shell BRD2703A chip_build_libshell=true
> ```
> 
> **Build error**
> 
> ```c++
> In file included from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/CodeUtils.h:34,
>                  from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/core/Optional.h:32,
>                  from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/app/ConcreteAttributePath.h:23,
>                  from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/zzz_generated/app-common/app-common/zap-generated/cluster-objects.h:26,
>                  from ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.h:20,
>                  from ../../../examples/refrigerator-app/silabs/include/EventHandlerLibShell.h:19,
>                  from ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:18:
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp: In function 'CHIP_ERROR RefrigeratorDoorEventHandler(int, char**)':
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:26: error: format '%d' expects argument of type 'int', but argument 4 has type 'chip::app::Cluster                s::RefrigeratorAlarm::AlarmBitmap' [-Werror=format=]
>   109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
>       |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~
>       |                                                                       |
>       |                                                                       chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap
> ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:469:70: note: in definition of macro 'ChipInternalLogImpl'
>   469 |             chip::Logging::Log(chip::Logging::kLogModule_##MOD, CAT, MSG, ##__VA_ARGS__);                                          \
>       |                                                                      ^~~
> ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:110:40: note: in expansion of macro 'ChipInternalLog'
>   110 | #define ChipLogProgress(MOD, MSG, ...) ChipInternalLog(MOD, PROGRESS, MSG, ##__VA_ARGS__)
>       |                                        ^~~~~~~~~~~~~~~
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:5: note: in expansion of macro 'ChipLogProgress'
>   109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
>       |     ^~~~~~~~~~~~~~~
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:109:61: note: format string is defined here
>   109 |     ChipLogProgress(Zcl, "Setting event for the door state %d", data->doorState);
>       |                                                            ~^
>       |                                                             |
>       |                                                             int
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp: In function 'void EventWorkerFunction(intptr_t)':
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:30: error: format '%d' expects argument of type 'int', but argument 4 has type 'chip::app::Cluster                s::RefrigeratorAlarm::AlarmBitmap' [-Werror=format=]
>   158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
>       |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~
>       |                                                                       |
>       |                                                                       chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap
> ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:469:70: note: in definition of macro 'ChipInternalLogImpl'
>   469 |             chip::Logging::Log(chip::Logging::kLogModule_##MOD, CAT, MSG, ##__VA_ARGS__);                                          \
>       |                                                                      ^~~
> ../../../examples/refrigerator-app/silabs/third_party/connectedhomeip/src/lib/support/logging/TextOnlyLogging.h:110:40: note: in expansion of macro 'ChipInternalLog'
>   110 | #define ChipLogProgress(MOD, MSG, ...) ChipInternalLog(MOD, PROGRESS, MSG, ##__VA_ARGS__)
>       |                                        ^~~~~~~~~~~~~~~
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:9: note: in expansion of macro 'ChipLogProgress'
>   158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
>       |         ^~~~~~~~~~~~~~~
> ../../../examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:158:56: note: format string is defined here
>   158 |         ChipLogProgress(Zcl, "Changing the door state %d", alarmData->doorState);
>       |                                                       ~^
>       |                                                        |
>       |                                                        int
> At global scope:
> cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
> cc1plus: all warnings being treated as errors
> [841/895] c++ obj/third_party/connectedhomeip/examples/platform/silabs/efr32-common.OTAConfig.cpp.o
> ```
> 
> [PR36548_2703.log](https://github.com/user-attachments/files/17822237/PR36548_2703.log)

Adds changes to fix this build error

**Comment by shgutte** [comment]
> @shgutte You'll have to reformat your code. 1 test is failing.

Hi @lboue reformated the code

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        ChipLogError(Shell, "Invalid argument: Input must be a valid integer.");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Convert the argument to an integer
+    int value = std::stoi(argv[0]); // Safe to use now, as we validated the input earlier
+
+    RefrigeratorAlarmEventData * data = Platform::New<RefrigeratorAlarmEventData>();
+    data->eventId                     = Events::Notify::Id;
+    data->doorState                   = static_cast<AlarmBitmap>(atoi(argv[0]));
```
Comment (jmartinez-silabs) [inline examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:117]: `value` of line 113 be used here now.

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        break;
+    }
+
+    default: {
+        ChipLogError(Zcl, "Invalid Event Id %s, line %d", __func__, __LINE__);
+        break;
+    }
+        // Free memory
+        Platform::Delete(data);
+    }
```
Comment (jmartinez-silabs) [inline examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp:177]: this shouldn't be in the switch { } well maybe it works but I never seen it done this way
```suggestion
    }
    
    // Free memory
    Platform::Delete(data);
```

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp (hunk 7 above)
Comment (shgutte) [reply examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp]: added this change

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**